lib/customer-deposit-wallet/http/Cardano/Wallet/Deposit/HTTP/Server.hs

```diff
@@ -90,17 +91,20 @@ createWalletViaMnemonic
     dir
     boot
     resource
-    (PostWalletViaMnemonic mnemonics' passphrase' users') =
-        onlyOnWalletIntance resource initWallet $> NoContent
-      where
-        initWallet :: WalletResourceM ()
-        initWallet =
-            REST.initWallet
-                tracer
-                boot
-                dir
-                (credentialsFromMnemonics mnemonics' passphrase')
-                (fromIntegral users')
+    (PostWalletViaMnemonic mnemonics' passphrase' users') = do
+        case mkMnemonic mnemonics' of
+            Left e -> fail $ show e
+            Right mnemonics'' -> do
```

Comment:
suggestion: rename `mnemonics''` to something a bit more descriptive, like `decodedMnemonics`? Double primes are prone to lead to annoying errors